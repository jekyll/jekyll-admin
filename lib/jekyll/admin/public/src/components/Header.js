import React, { Component, PropTypes } from 'react';
import { Link } from 'react-router';

import styles from '../styles/header.scss';

const Header = ({title}) => {
  return (
    <div className="header-container">
      <h3 className="title">
        <Link target="_blank" to={`/`}><i className="fa fa-home"></i> <span>{title}</span></Link>
      </h3>
    </div>
  );
};

Header.propTypes = {
  title: PropTypes.string.isRequired
};

export default Header;
