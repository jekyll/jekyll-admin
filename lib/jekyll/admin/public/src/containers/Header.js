import React, { Component, PropTypes } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { Link } from 'react-router';
import _ from 'underscore';

// Actions
import { fetchConfig } from '../actions/config';

export class Header extends Component {

  componentDidMount() {
    const { fetchConfig } = this.props;
    fetchConfig();
  }

  render() {
    const { config } = this.props;
    let title = _.isEmpty(config) ? '' : config.title;
    return (
      <div className="header">
        <h3 className="title">
          <Link target="_blank" to={`/`}>
            <i className="fa fa-home"></i> <span>{title}</span>
          </Link>
        </h3>
      </div>
    );
  }
}

Header.propTypes = {
  fetchConfig: PropTypes.func.isRequired,
  config: PropTypes.object.isRequired
};

function mapStateToProps(state) {
  const { config } = state;
  return {
    config: config.config,
    isFetching: config.isFetching
  };
}

function mapDispatchToProps(dispatch) {
  return bindActionCreators({
    fetchConfig
  }, dispatch);
}

export default connect(mapStateToProps, mapDispatchToProps)(Header);
