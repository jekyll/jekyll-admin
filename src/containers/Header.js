import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router';
import Icon from '../components/Icon';

export class Header extends Component {
  render() {
    const { config } = this.props;
    return (
      <div className="header">
        <h3 className="title">
          <Link target="_blank" to={config.url || '/'}>
            <Icon name="home" />
            <span>{config.title || 'You have no title!'}</span>
          </Link>
        </h3>
        <span className="version">v{process.env.REACT_APP_VERSION}</span>
      </div>
    );
  }
}

Header.propTypes = {
  config: PropTypes.object.isRequired,
};

export default Header;
