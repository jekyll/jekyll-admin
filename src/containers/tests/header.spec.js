import React from 'react';
import { mount } from 'enzyme';
import { Header } from '../Header';
import { config } from './fixtures';

const defaultProps = { config: config.content };

function setup(props = defaultProps) {
  const component = mount(
    <Header {...props} />
  );

  return {
    component: component,
    title: component.find('h3 span')
  };
}

describe('Containers::Header', () => {
  it('should render correctly', () => {
    const { component, title } = setup();
    const { config } = component.props();
    expect(title.text()).toEqual(config.title);
  });

  it('should render placeholder title', () => {
    const { component, title } = setup(Object.assign({}, defaultProps, {
      config: {}
    }));
    const { config } = component.props();
    expect(title.text()).toEqual('You have no title!');
  });
});
