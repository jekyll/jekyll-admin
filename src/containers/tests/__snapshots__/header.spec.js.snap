// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Containers::Header renders correctly 1`] = `
<div
  className="header"
>
  <h3
    className="title"
  >
    <a
      onClick={[Function]}
      style={Object {}}
      target="_blank"
    >
      <i
        aria-hidden="true"
        className="fa fa-home"
      />
      <span>
        Your awesome title
      </span>
    </a>
  </h3>
  <span
    className="version"
  >
    v
  </span>
</div>
`;

exports[`Containers::Header renders placeholder title 1`] = `
<div
  className="header"
>
  <h3
    className="title"
  >
    <a
      onClick={[Function]}
      style={Object {}}
      target="_blank"
    >
      <i
        aria-hidden="true"
        className="fa fa-home"
      />
      <span>
        You have no title!
      </span>
    </a>
  </h3>
  <span
    className="version"
  >
    v
  </span>
</div>
`;
